### DISCUSSION TITLE
Source maps and autoLabel

### DISCUSSION THREAD

**Discussion by isshesure** [system-ui/theme-ui#1932] [Q&A] [answered]
Is there a recommended setup to improve debugging with Theme UI? 

I'm using the `sx` prop and inline styles (rather than styled components) as recommended at https://theme-ui.com/guides/styled-components#using-the-sx-prop, and the class names are e.g. `css-zdrrq2` etc.

I have added the [babel emotion plugin](https://github.com/emotion-js/emotion/tree/main/packages/babel-plugin) with source maps and `autoLabel` enabled in dev (default options), but I'm not seeing any source maps added in the output and auto labelling does not appear to work. 

I can add a `label` manually to each `sx` usage, and that is added to the output class name as expected, however it would be beneficial for the `autoLabel` feature of the babel emotion plugin to work, and to be able to navigate to the source code from each inline style?

**Comment by lachlanjc** [comment, marked as answer]
This isn't a feature we support afaik, but sounds great, so PRs welcome as always!
